test/test_array_ops.cpp

```diff
@@ -157,6 +157,40 @@ TEST(ArrayOps, Unpack) {
   }  // end of for loop
 }  // end of testing Unpack
 
+TEST(ArrayOps, Fill) {
+  vector<int> static_input_indexes = {0};  // has static input
+
+  Scope root1 = Scope::NewRootScope();
+
+  // 0-D(scalar) value to fill the returned tensor
+  Tensor input_data(DT_FLOAT, TensorShape({}));
+  AssignInputValuesRandom(input_data);
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  // Fill creates a tensor filled with scalar value
+  // 1-D shape of the output tensor
+  auto shape1 =
+      ops::Const(root1, {int32(2), int32(1), int32(3)}, TensorShape({3}));
+  auto R1 = ops::Fill(root1, shape1, input_data);
```

Comment:
So earlier we ran into the problem of incorrect types and casting. You seem to have figured out the right types to use here ```int32(2)```:

So now you can have a loop over different sizes.
```
vector<vector<i32>> shape_vector= {{2,1,3}, {1,2}, {4,5}};
...
for (auto shape : shape_vector)
{
  auto shape_const = ops::Const(root1, shape, TensorShape({shape.size()});
  ...
}
```

Reply:
As discussed, you are right, we cannot use vectors. Please checkout slack for a workaround

Reply:
I did try to use the input_sizes vector and loop for different sizes but that did not work as Fill op is not taking pre-defined vectors. So used constant and assigned int32 values to the tensor.